pkg/kubelet/cm/topologymanager/policy_best_effort.go

```diff
@@ -39,9 +49,31 @@ func (p *bestEffortPolicy) canAdmitPodResult(hint *TopologyHint) bool {
 	return true
 }
 
+func (p *bestEffortPolicy) preferClosestNUMAFunc(candidate *TopologyHint, current *TopologyHint) bool {
+	if candidate.NUMANodeAffinity.Count() == current.NUMANodeAffinity.Count() {
+		candidateDistance := p.numaTopology.CalculateAvgDistanceFor(candidate.NUMANodeAffinity)
+		currentDistance := p.numaTopology.CalculateAvgDistanceFor(current.NUMANodeAffinity)
+		// if average distance is the same take bitmask with more lower-number bits set
+		if candidateDistance == currentDistance {
+			return candidate.NUMANodeAffinity.IsLowerValueThan(current.NUMANodeAffinity)
+		}
+		// candidate avg distance is lower
+		return candidateDistance < currentDistance
+	}
+
+	return candidate.NUMANodeAffinity.IsNarrowerThan(current.NUMANodeAffinity)
+}
+
 func (p *bestEffortPolicy) Merge(providersHints []map[string][]TopologyHint) (TopologyHint, bool) {
+	var fitnessFunc fitterHintFunc
+
+	if p.opts.PreferClosestNUMA {
+		fitnessFunc = p.preferClosestNUMAFunc
+	}
+
```

Comment:
It would be better to be explicit about the value of fitnessFunc even in the default case here, so something like:

```
fitnessFunc := p.preferNarrowerHintFunc
if p.opts.PreferClosestNUMA {
	fitnessFunc = p.preferClosestNUMAFunc
}
```